src/ExpressionEvaluator/Core/Source/ResultProvider/Helpers/TypeHelpers.cs

```diff
@@ -389,6 +389,14 @@ internal static bool IsTupleCompatible(this Type type, out int cardinality)
             return false;
         }
 
+        // Returns cardinality if tuple type, otherwise 0.
+        internal static int GetTupleCardinalityIfAny(this Type type)
```

Comment:
There is currently no way to create a 0-tuple. It cannot be specified in source, it is not created internally, and public APIs block it too.